File: examples/LPT - LCD/ProgrammingLPT/LPT/iosample.2009-07-15/net2005/cpp/iodemo.cpp

/*
	iodemo.cpp : CDC-IO demo

	O.Tamura @ Recursion Co., Ltd.
*/

#include "atmega48.h"		//	ATmega48/88/168
//#include "atmega8.h"		//	ATmega8

#include "cdcio.h"


#if 1
int main(array<System::String ^> ^args)
{
	CDCIO		sfr;
	int			i, vc, vn, t;
	String		^bar = "**********************************************************************";


	if( args->Length==0 ) {
		Console::WriteLine( " usage: iodemo COM_port_number" );
		return -1;
	}
	//	Open COM device
	if( !sfr.Open( args[0] ) ) {
		Console::WriteLine( " COM{0} is not available.", args[0] );
		return -2;
	}

    Console::WriteLine( sfr.Who() );

	//	Initialize AVR peripherals
	sfr.Set( PORTB, 0 );
	sfr.Set( DDRB, 0x04 );		//	OUT: PB2
	sfr.Set( PORTC, 0 );
	sfr.Set( DDRC, 0x0f );		//	OUT: PC3-0, IN: PC4

	sfr.Set( TCCR1A, 0x10 );

#ifdef _ATMEGA48_
	sfr.Set( DIDR0, 0x10 );
	sfr.Set( ADCSRB, 0 );
#endif
	sfr.Set( ADMUX, 0x04 );
	sfr.Set( ADCSRA, 0x87 );

	i	= 0;
	vc	= 0;
	while( !Console::KeyAvailable ) {

		//	A/D Conversion
		sfr.SetOr( ADCSRA, 0x40 );		//	Start conversion
		Thread::Sleep( 10 );			//	wait 10mSec
		vn	= sfr.Get( ADCL );			//	Obtain the result
		vn	|= sfr.Get( ADCH ) << 8;
		if( (vc-1)<=vn && vn<=(vc+1) ) {
			//	Flip LEDs if the ADC value is not changed
			vc	= vn;
			Thread::Sleep( 300 );
			sfr.SetXor( PORTC, 0x0f );
			sfr.Set( TCCR1B, 0 );
			continue;
		}
		vc	= vn;

		//	Light LEDs with the ADC value
		sfr.Set( PORTC, 8>>(vc>>8) );

		//	Change buzzer tone with the ADC value
		t	= ( vc + 0x100 ) << 4;
		sfr.Set( OCR1AH, t>>8 );
		sfr.Set( OCR1AL, t );
		sfr.Set( TCCR1B, 0x09 );

		//	Display the scale with the ADC value
		Console::WriteLine( String::Format("{0,6:d}  {1}", vc, bar->Substring(0,vc>>4) ) );

		Thread::Sleep( 20 );
	}

	//	Reset AVR peripherals
	sfr.Set( ADCSRA, 0 );
#ifdef _ATMEGA48_
	sfr.Set( DIDR0, 0 );
#endif
	sfr.Set( TCCR1A, 0 );
	sfr.Set( TCCR1B, 0 );
	sfr.Set( PORTC, 0 );
	sfr.Set( DDRC, 0 );
	sfr.Set( PORTB, 0 );
	sfr.Set( DDRB, 0 );

	sfr.Close();
    return 0;
}

#else
//	Read/Write EEPROM data
int main(array<System::String ^> ^args)
{
	CDCIO		sfr;
	int			i;


	if( args->Length==0 ) {
		Console::WriteLine( " usage: sfrdemo COM_port_number" );
		return -1;
	}
	//	Open COM device
	if( !sfr.Open( args[0] ) ) {
		Console::WriteLine( " COM{0} is not available.", args[0] );
		return -2;
	}

    Console::WriteLine( sfr.Who() );

#if 1
	//	Write
    Console::Write( "\n EEPROM write" );
	sfr.Set( EECR, 0 );
	for( i=0; i<256; i++ ) {
		if( (i&0x3f)==0 )
			Console::Write( "\n\t" );
		sfr.Set( EEARL, i );
//		sfr.Set( EEDR, i );
		sfr.Set( EEDR, ~i );
		sfr.Set2( EECR, 4, 6 );
		Thread::Sleep( 10 );			//	wait 10mSec
		Console::Write( "." );
	}
#endif
	//	Read
    Console::Write( "\n EEPROM read" );
	for( i=0; i<256; i++ ) {
		if( (i&0x0f)==0 )
			Console::Write( "\n\t" );
		sfr.Set( EEARL, i );
		sfr.Set( EECR, 1 );
		Console::Write( " {0:x2}", sfr.Get( EEDR ) );
	}
	Console::WriteLine( "" );

	sfr.Close();
	return 0;
}
#endif

File: examples/LPT - LCD/ProgrammingLPT/LPT/iosample.2009-07-15/vs2003/vb/iodemo.vb

'   CDC-IO demo for Visual Basic

'       Osamu Tamura @ Recursion Co., Ltd.

Imports System
Imports System.Threading
Imports System.Text
Imports System.Globalization
Imports System.Runtime.InteropServices


Module iodemo
    Private _kbhit As Boolean

    Private Sub KeyEnter()
        Console.Read()
        _kbhit = True
    End Sub

    Sub Main(ByVal CmdArgs() As String)
        Dim Quit As Thread = New Thread(New ThreadStart(AddressOf KeyEnter))
        Dim sfr As CDCIO
        Dim s As String
        Dim vc, vn, t As Integer
        Dim bar = "************************************************************************"

        If CmdArgs.Length = 0 Then
            Console.WriteLine(" usage: usfrdemo COM_port_number")
            Return
        End If

        '	Open COM device
        sfr = New CDCIO
        If Not sfr.Open(CmdArgs(0)) Then
            Console.WriteLine(" COM{0} is not available.", CmdArgs(0))
            Return
        End If

        Console.WriteLine(sfr.Who())

        '	 Initialize AVR peripherals
        sfr.SetValue(AVRSFR.PORTB, 0)
        sfr.SetValue(AVRSFR.DDRB, &H4)      '	OUT: PB2
        sfr.SetValue(AVRSFR.PORTC, 0)
        sfr.SetValue(AVRSFR.DDRC, &HF)      '	OUT: PC3-0, IN: PC4

        sfr.SetValue(AVRSFR.TCCR1A, &H10)

        sfr.SetValue(AVRSFR.DIDR0, &H10)    '   ATmega8 :: REMOVE this line
        sfr.SetValue(AVRSFR.ADMUX, &H4)
        sfr.SetValue(AVRSFR.ADCSRB, 0)      '   ATmega8 :: REMOVE this line
        sfr.SetValue(AVRSFR.ADCSRA, &H87)

        vc = 0
        _kbhit = False
        Quit.IsBackground = True
        Quit.Start()
        While Not _kbhit
            '	A/D Conversion
            sfr.SetOr(AVRSFR.ADCSRA, &H40)  '  Start conversion
            Thread.Sleep(10)                '  wait 10mSec
            vn = sfr.GetValue(AVRSFR.ADCL)  '  Obtain the result
            vn = vn + (sfr.GetValue(AVRSFR.ADCH) << 8)
            If (vc - 1) <= vn And vn <= (vc + 1) Then
                '	Flip LEDs if the ADC value is not changed
                vc = vn
                Thread.Sleep(300)
                sfr.SetXor(AVRSFR.PORTC, &HF)
                sfr.SetValue(AVRSFR.TCCR1B, 0)
            Else
                vc = vn

                '	Light LEDs with the ADC value
                sfr.SetValue(AVRSFR.PORTC, 8 >> (vc >> 8))

                '	Change buzzer tone with the ADC value
                t = (vc + &H100) << 4
                sfr.SetValue(AVRSFR.OCR1AH, t >> 8)
                sfr.SetValue(AVRSFR.OCR1AL, t)
                sfr.SetValue(AVRSFR.TCCR1B, &H9)

                '	Display the scale with the ADC value
                Console.WriteLine(String.Format("{0,6:d}  {1}", vc, bar.Substring(0, vc >> 4)))

                Thread.Sleep(20)
            End If
        End While

        '	Reset AVR peripherals
        sfr.SetValue(AVRSFR.ADCSRA, 0)
        sfr.SetValue(AVRSFR.DIDR0, 0)       '   ATmega8 :: REMOVE this line
        sfr.SetValue(AVRSFR.TCCR1A, 0)
        sfr.SetValue(AVRSFR.TCCR1B, 0)
        sfr.SetValue(AVRSFR.PORTC, 0)
        sfr.SetValue(AVRSFR.DDRC, 0)
        sfr.SetValue(AVRSFR.PORTC, 0)
        sfr.SetValue(AVRSFR.DDRB, 0)
        sfr.SetValue(AVRSFR.PORTB, 0)

        sfr.Close()
    End Sub

End Module
